crypto/ake/enclave/src/lib.rs

```diff
@@ -438,24 +440,43 @@ impl<EI: EnclaveIdentity> AkeEnclaveState<EI> {
     }
 
     /// Transforms an incoming client message, i.e. a message sent from a client
-    /// to the current enclave, into a list of outbound  messages for
-    /// other enclaves that serve as backends to the current enclave.
-    ///
-    ///                               / --> Backend Enclave 1
-    ///   Client -> Current Enclave ---> Backend Enclave 2
-    ///                              \ --> Backend Enclave N
-    pub fn reencrypt_client_message_for_backends(
+    /// to the current enclave, into a sealed message which can be decrypted
+    /// later for use by this enclave without advancing the Noise nonce.
+    pub fn decrypt_client_message_for_enclave(
         &self,
         incoming_client_message: EnclaveMessage<ClientSession>,
+    ) -> Result<SealedClientMessage> {
+        let aad = incoming_client_message.aad.clone();
+        let channel_id = incoming_client_message.channel_id.clone();
+        let client_query_bytes = self.client_decrypt(incoming_client_message)?;
+        let sealed_data =
+            IntelSealed::seal_raw(&client_query_bytes, &[]).map(|x| x.as_ref().to_vec())?;
```

Comment:
IIUC, this method not only converts the `client_query_bytes` into a seal message, but also stores the sealed message in the enclave? If this is accurate, can you update the methods comment to reflect this?

If so, does the `sealed_data` represent a "handle" or a "pointer" to the actual data that's being sealed? If so, can we rename to something like `sealed_data_handle`? I'd want to make it clear that the thing that's being passed back across the enclave boundary isn't the data itself but some sort of identifier.

Reply:
My understanding is that the sealed data doesn't live in the enclave, instead the sealing is done by encrypting with a key stored in the enclave and returning the ciphertext. I believe `IntelSealed` contains the full ciphertext. Maybe @jcape can clear this up for us.